### PR TITLE
ui: Check for `intention` ACL resources, not `intentions` ACL resources

### PR DESCRIPTION
**Pull request by johncowen**
TLDR Remove an additional `s`.

---

ACL intention rules are called `intention` not `intentions`. 

The code involved here temporarily ignores the result of our HTTP API result as it does not function exactly how we need (more info in this issue but _not_ essential reading for PR purposes: https://github.com/hashicorp/consul/pull/11520), the original PR to do this ignore is https://github.com/hashicorp/consul/pull/11520.

The ignore essentially loops through the response of the HTTP request we make to the auth HTTP API and overwrites the ones which could be wrong (only a subset hence the `includes`) with a hardcoded `Allow: true`, which means we still show UI links for places that, once navigated or submitted to, may result in a 403 response from the backend.

The original desired UX improvement was to not even show the links if you weren't able to perform actions or visit those pages in the first place.

As we had an additional `s` on `intention` here, it wasn't overwriting any potentially erroneous intention ACL rules, and therefore, when they were erroneous the link in the left hand side navigation for intentions would not be visible.

Here's the reference in the code to the list of rules we request, note `intention`:

https://github.com/hashicorp/consul/blob/dd1179177fc8303c8ca3ee84c4cd28182803d4ad/ui/packages/consul-ui/app/services/repository/permission.js#L44-L50

Finally this fix (so the one from the PR before this is temporary https://github.com/hashicorp/consul/pull/11520) and will be removed as soon as the HTTP API functions as desired. The previous PR has not yet landed in a release, hence no changelog here, but autobackports are required for 1.10 and 1.11 and the necessary labels have been added here.

There is currently a CI test fail here, which reading the title of it is actually expected with this change, will look and fix up now.

### CODE REVIEW

**Review by amyrlam — COMMENTED** [review]
Seems like the test is still failing, ok to +1 once it's fixed. Will add to Mon's calendar event

**Comment by johncowen** [comment]
This ones fixed up, the fail made sense as we hardcode/overwrite the `write: false` to `write:true` now from within the app. I added comments to the test here also for future us in case we need a reminder that this is ok to simply uncomment again when the time comes.

**Review by natmegs — APPROVED** [review]
Pair reviewed

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/534689.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 72cddf4f49c6419f9370c19e3fabca415a21d3b4 onto `release/1.11.x` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 72cddf4f49c6419f9370c19e3fabca415a21d3b4 onto `release/1.10.x` succeeded!
